src/openrct2/world/Location.hpp

```diff
@@ -254,6 +254,11 @@ struct TileCoordsXY
 
     CoordsXY ToCoordsXY() const
     {
+        if (isNull())
+        {
+            return { COORDS_NULL, 0 };
```

Comment:
I'd prefer if TileCoordsXY didn't handle implementation details of CoordsXY. CoordsXY should handle setting it to null itself.

Reply:
This has now been changed. I was going to create another method that returns a null from CoordsXY but decided that it would just make it more annoying when we come to remove all of the nulls.